src/os/shared.rs

```diff
@@ -106,6 +106,9 @@ pub fn get_input(
         .map(|iface| get_datalink_channel(iface))
         .filter_map(Result::ok)
         .collect::<Vec<_>>();
+    if network_frames.len() == 0 {
+        failure::bail!("Could not find any network interface to listen to. Try running with sudo");
```

Comment:
Yes, this was the missing part indeed!

Now that we have the information that we could not listen due to a permission error (since #59), maybe we could check that this happened before showing this message? Otherwise it could be misleading, specially if already running with elevated privileges.

I would also suggest using a more generic message that avoids speaking about sudo given that sudo may not be installed in the target system. What do you think about "try with elevated privileges"? If we do that we should also remove the reference to sudo in `get_datalink_channel`

Reply:
Right - fixed. :)

As for sudo... I definitely understand what you're saying, but I feel it's more accessible to new users this way. What do you say?

Reply:
I see your point, maybe we could reformulate it but I'm not going to nitpick on that :)

We can maybe change it when there is Windows support ;)

Reply:
Sounds good. I know traditionally such tools would say "permission error" and let you figure it out, but I really want to try for a "do what I need and tell me what to do" experience here. :)

Thanks for the review! Maybe we can open a separate issue for proper error handling in the OS layer? Would help with debugging, certainly. Maybe do something like this: https://rust-lang-nursery.github.io/failure/error-errorkind.html